whois-api/src/main/java/net/ripe/db/whois/api/httpserver/JettyBootstrap.java

```diff
@@ -223,7 +223,7 @@ private Connector createConnector(final Server server) {
         final HttpConfiguration httpConfiguration = new HttpConfiguration();
         if (isHttpProxy()) {
             // client address is set in X-Forwarded-For header by HTTP proxy
-            httpConfiguration.addCustomizer(new RemoteAddressCustomizer());
+            httpConfiguration.addCustomizer(new RemoteAddressCustomizer(trustedIpRanges));
```

Comment:
This `if` means that the `RemoteAddressCustomizer` is only used if the LB proxies the request (i.e. HTTPS is *off* in the backend).

If you want that class to always handle 'clientIp` then you need to move the LB check there also.

Reply:
Thanks for pointing this out I missed this